frontend_tests/node_tests/subs.js

```diff
@@ -129,135 +123,73 @@ run_test("filter_table", (override) => {
 
     // Search names AND descriptions
-    sub_table_append = [];
-
-    subs.filter_table({input: "Mon", subscribed_only: false});
-    assert($(".stream-row-denmark").hasClass("notdisplayed"));
-    assert(!$(".stream-row-poland").hasClass("notdisplayed"));
-    assert(!$(".stream-row-pomona").hasClass("notdisplayed"));
-    assert($(".stream-row-cpp").hasClass("notdisplayed"));
-    assert($(".stream-row-zzyzx").hasClass("notdisplayed"));
-    assert.deepEqual(sub_table_append, [
-        ".stream-row-pomona",
-        ".stream-row-poland",
-        ".stream-row-cpp",
-        ".stream-row-zzyzx",
-        ".stream-row-denmark",
-    ]);
+    test_filter({input: "Mon", subscribed_only: false}, [pomona, poland]);
 
     // Explicitly order streams by name
-    sub_table_append = [];
-    subs.filter_table({input: "", subscribed_only: false, sort_order: "by-stream-name"});
-    assert.deepEqual(sub_table_append, [
-        ".stream-row-cpp",
-        ".stream-row-denmark",
-        ".stream-row-poland",
-        ".stream-row-pomona",
-        ".stream-row-zzyzx",
+    test_filter({input: "", subscribed_only: false, sort_order: "by-stream-name"}, [
+        cpp,
+        denmark,
+        poland,
+        pomona,
+        zzyzx,
     ]);
 
     // Order streams by subscriber count
-    sub_table_append = [];
-    subs.filter_table({input: "", subscribed_only: false, sort_order: "by-subscriber-count"});
-    assert.deepEqual(sub_table_append, [
-        ".stream-row-poland",
-        ".stream-row-cpp",
-        ".stream-row-zzyzx",
-        ".stream-row-denmark",
-        ".stream-row-pomona",
+    test_filter({input: "", subscribed_only: false, sort_order: "by-subscriber-count"}, [
+        poland,
+        cpp,
+        zzyzx,
+        denmark,
+        pomona,
     ]);
 
     // Order streams by weekly traffic
-    sub_table_append = [];
-    subs.filter_table({input: "", subscribed_only: false, sort_order: "by-weekly-traffic"});
-    assert.deepEqual(sub_table_append, [
-        ".stream-row-poland",
-        ".stream-row-cpp",
-        ".stream-row-zzyzx",
-        ".stream-row-pomona",
-        ".stream-row-denmark",
+    test_filter({input: "", subscribed_only: false, sort_order: "by-weekly-traffic"}, [
+        poland,
+        cpp,
+        zzyzx,
+        pomona,
+        denmark,
     ]);
 
-    // Showing subscribed streams only puts un-subscribed DOM elements last
-    sub_table_append = [];
-    subs.filter_table({input: "", subscribed_only: true, sort_order: "by-subscriber-count"});
-    assert.deepEqual(sub_table_append, [
-        ".stream-row-poland",
-        ".stream-row-cpp",
-        ".stream-row-zzyzx",
-        ".stream-row-pomona",
-        ".stream-row-denmark",
+    // Sort for subscribed only.
+    test_filter({input: "", subscribed_only: true, sort_order: "by-subscriber-count"}, [
+        poland,
+        cpp,
+        zzyzx,
+        pomona,
     ]);
```

Comment:
This list seems to not have Denmark in the expected list, in comparison to the previous version; what's the story there?

Reply:
Is the issue that the previous test code was buggy, as the user isn't subscribed to `Denmark`?  Or are we just no longer verifying that it's present-but-hidden?

Reply:
The test code was noting, via `sub_table_append`, that Denmark was the last stream we were inserting into the DOM, which is still correct on master.  But it's an overly specific implementation detail to test; even though Denmark gets hackily re-sorted into the DOM, we don't actually show it.

Now the tests just verify the final sorted, filtered list of streams.  We cheat now by asking `filter_table` to just tell us what it's confusingly producing as its final outcome, but a subsequent change will make it so we don't have to cheat quite as much.

Reply:
Great, sounds like this makes the test better.